### DISCUSSION TITLE
How to clear old deployment cache and serviceworker issue?

### DISCUSSION THREAD

**Discussion by 2manoj1** [vercel/vercel#3956] [General] [answered]
Previously I deploy next.js PWA app (next-pwa) but its not updating api data. So I remove that code and redeploy but its not clear all cache in all my client browser. 

` "next": "9.3.1",
"next-pwa": "^2.3.1",
"react": "16.13.1",
    "react-dom": "16.13.1"`

Deploy With `Now`

Help me how clear cache?

**Comment by paulogdm** [comment, marked as answer]
Hello,

You can't control nor purge a "client cache". For that reason, I recommend that you never cache anything in the client unless something is really immutable, like images or hashed bundles.
This problem is outside the scope of the platform since it is a browser feature.

**Reply by styfle** [reply]
The best way to handle this, is  cache at the CDN level instead using `s-maxage` headers.

Read more about caching in the documentation here: https://zeit.co/docs/v2/network/caching
